plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetDetails.tsx

```diff
@@ -36,8 +39,19 @@ const DatasetDetails: React.FC<Props> = ({
 
   const handleDatasetAdd = useCallback(() => {
     if (!dataset || addDisabled) return;
+    dataset.layers = layers;
```

Comment:
This is modifying the dataset prop directly, no? It is preferable to create a new variable to pass in to the onDatasetAdd